apps/server/src/modules/authentication/services/authentication.service.ts

```diff
@@ -131,4 +147,65 @@ export class AuthenticationService {
 	public normalizePassword(password: string): string {
 		return password.trim();
 	}
+
+	public async logoutFromExternalSystem(userId: string, systemId: string): Promise<void> {
+		if (!this.configService.get<boolean>('FEATURE_EXTERNAL_SYSTEM_LOGOUT_ENABLED')) {
+			throw new ExternalSystemLogoutIsDisabledLoggableException();
+		}
+
+		const system: System | null = await this.systemService.findById(systemId);
+		const sessionToken: OauthSessionToken | null = await this.oauthSessionTokenService.findLatestByUserId(userId);
+
+		if (!sessionToken || !system) {
+			return;
+		}
+
+		if (!system.oauthConfig) {
+			throw new OauthConfigMissingLoggableException(systemId);
+		}
+
+		const now = new Date();
+		if (now > sessionToken.expiresAt) {
+			await this.oauthSessionTokenService.delete(sessionToken);
+			return;
+		}
+
+		if (!system?.oauthConfig.endSessionEndpoint) {
+			throw new EndSessionEndpointNotFoundLoggableException(systemId);
+		}
+
+		const headers: AxiosHeaders = new AxiosHeaders();
+		headers.setContentType('application/x-www-form-urlencoded');
+
+		const config: AxiosRequestConfig = {
+			auth: {
+				username: system?.oauthConfig.clientId,
+				password: this.oauthEncryptionService.decrypt(system?.oauthConfig.clientSecret),
+			},
+			headers,
+		};
+
+		try {
+			await firstValueFrom(
+				this.httpService.post(
+					system?.oauthConfig?.endSessionEndpoint,
+					{
+						refresh_token: sessionToken.refreshToken,
+					},
+					config
+				)
+			);
+		} catch (err) {
+			const axiosError = err as AxiosError;
+			let errorResponseData: string;
+			if (axiosError.response) {
+				errorResponseData = JSON.stringify({ code: axiosError.code, response: axiosError.response.data });
+			} else {
+				errorResponseData = JSON.stringify({ code: axiosError.code, response: axiosError.message });
+			}
```

Comment:
Message formatting should be done inside the loggable. Maybe look into the AxiosErrorLoggable as a reference.